meteor/client/ui/TestTools/RecordingsList.tsx

```diff
@@ -0,0 +1,290 @@
+}
+
+const RecordingsList = translateWithTracker<IRecordingListProps, IRecordingListState, IRecordingListTrackedProps>((props: IRecordingListProps) => {
+	return {
+		studio: StudioInstallations.findOne(),
+		files: RecordedFiles.find({}, { sort: { startedAt: -1 } }).fetch()
+	}
+})(class RecordedFilesList extends MeteorReactComponent<Translated<IRecordingListProps & IRecordingListTrackedProps>, IRecordingListState> {
+
+	constructor (props: Translated<IRecordingListProps & IRecordingListTrackedProps>) {
+		super(props)
+
+		this.state = {
+			filename: '',
+			showDeleteConfirm: false
+		}
+	}
+
+	onUpdateValue = (edit: any, newValue: any) => {
+		console.log('edit', edit, newValue)
+		let attr = edit.props.attribute
+
+		if (attr) {
+			let m = {}
+			m[attr] = newValue
+			this.setState(m)
+		}
+	}
+
+	componentWillMount () {
+		if (this.props.match && this.props.match.params) {
+			// Subscribe to data:
+			this.subscribe('recordedFiles', {
+				studioId: this.props.match.params.studioId
+			})
+			this.subscribe('studioInstallations', {
+				_id: this.props.match.params.studioId
+			})
+		}
+	}
+
+	stopRecording (e) {
+		if (this.props.match && this.props.match.params) {
+			Meteor.call(ClientAPI.methods.execMethod, eventContextForLog(e), TestToolsAPI.methods.recordStop, this.props.match.params.studioId, (err, res) => {
+				if (err || (res && res.error)) {
+					console.error(err || res)
+					// this.handleActivationError(err || res)
+					return
+				}
+			})
+		}
+	}
+	startRecording (e) {
+		if (this.props.match && this.props.match.params) {
+			Meteor.call(ClientAPI.methods.execMethod, eventContextForLog(e), TestToolsAPI.methods.recordStart, this.props.match.params.studioId, this.state.filename, (err, res) => {
+				if (err || (res && res.error)) {
+					console.error(err || res)
+					// this.handleActivationError(err || res)
+					return
+				}
+			})
+			this.setState({
+				filename: ''
+			})
+		}
+	}
+
+	handleConfirmDeleteCancel = (e) => {
+		this.setState({
+			deleteConfirmItem: undefined,
+			showDeleteConfirm: false
+		})
+	}
+	onDelete (item: RecordedFile) {
+		this.setState({
+			deleteConfirmItem: item,
+			showDeleteConfirm: true
+		})
+	}
+	handleConfirmDeleteAccept = (e) => {
+		if (this.state.deleteConfirmItem) {
+			Meteor.call(ClientAPI.methods.execMethod, eventContextForLog(e), TestToolsAPI.methods.recordDelete, this.state.deleteConfirmItem._id)
+		}
+		this.setState({
+			showDeleteConfirm: false
+		})
+	}
+
+	isStudioConfigured () {
+		const { studio } = this.props
+		if (!studio) return false
+
+		const config = objectPath.get(studio, 'testToolsConfig.recordings')
+		if (!config) return false
+
+		if (!config.channelIndex || !config.decklinkDevice || !config.deviceId) return false
+
+		return true
+	}
+
+	renderControlPanel () {
+		const { t } = this.props
+
+		if (!this.isStudioConfigured()) {
+			return <React.Fragment>
+				<p>{t('A required setting is not configured')}</p>
+			</React.Fragment>
```

Comment:
No need to return a `React.Fragment`, just return the `p`. `React.Fragment` is only needed when returning multiple, same-depth elements.